parser/src/lib.rs

```diff
@@ -235,6 +235,16 @@ public out = y(%last_row);"#;
             assert_eq!(input.trim(), printed.trim());
         }
 
+        #[test]
+        fn reparse_arrays() {
+            let input = "pol commit y[3];\n(y - 2) = 0;\n(y[2] - 2) = 0;\npublic out = y[1](2);";
```

Comment:
what does `(y - 2)` mean? Is it universally quantified?

Reply:
This just parses, so yes, it should break at some later type checking phase.

Reply:
ok